feature/Remove iface status update

morantr opened the pull request:
The "conf_initialized" is responsible for reading the common configurations only
once as long there was no change in the operational state.
Remove this flag because it solves some it causing some issues which the configuration
is not being updated if there is no change in the operational flag on the platform
manager and it allows removing a lot of irrelevant code which responsible
on updating the bpl on the ifaces status.
For now, we will read the common configuration twice but when we will move to
unified agent, we will read it only once.
See issue: #435

Since we are not depended on the iface status and it is not being used anymore, remove it completely from the code.

morantr committed:
common: Prevent killing of log tail when stopping prplMesh

Currently, the "killall_program" kills all running processes which contain the program name. This means that the log tail will also be killed.
Add some regex to the "pgrep" that eliminates the tail process.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove print that clatters the log

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

kantera800 commented on CMakeLists.txt:
```diff
@@ -107,7 +107,7 @@ set(BEEROCKS_BRIDGE_IFACE "br-lan")
     set(BEEROCKS_BH_WIRE_IFACE "eth1")
 endif()
-set(BEEROCKS_NOTIFY 0)
+set(BEEROCKS_NOTIFY 1)
```
not sure if this configuration is needed , i think it was used in dummy mode before we had hostapd .
If you decide keeping at least change it to BEEROCKS_IFACE_NOTIFY

tomereli replied:
This is was used to notify the beerocks SL - this is no longer needed, so why did you add it? It is not clear.

morantr replied:
The platform manager needs to know the status. It uses it.

morantr replied:
Removed completly

tomereli commented on common/beerocks/scripts/prplmesh_utils.sh.in:
```diff
@@ -33,7 +33,7 @@ killall_program() {
         KILL_SIG=$SIG_KILL
     fi
-    for PID in $(pgrep -f $PROGRAM_NAME); do
+    for PID in $(pgrep -f $PROGRAM_NAME[[:blank:][:alpha:]]); do
```
Oh yes! Thank you for this commit :)

tomereli commented on framework/platform/bpl/uci/cfg/bpl_cfg.cpp:
```diff
@@ -192,7 +192,7 @@ int cfg_get_wifi_params(const char iface[BPL_IFNAME_LEN], struct BPL_WLAN_PARAMS
     wlan_params->channel = 0;
     if (!cfg_get_channel(iface, &wlan_params->channel)) {
-        MAPF_INFO("UCI: interface " << iface << ": channel is not configured assuming auto\n");
+        // MAPF_INFO("UCI: interface " << iface << ": channel is not configured assuming auto\n");
```
So why not delete this print completely?

morantr replied:
I left it so if it is needed, for some debug session it could be easily added back. But I can remove it if you want.

morantr replied:
removed

tomereli reviewed (changes requested):
LGTM if you remove the commit which adds back NOTIFY to BPL which is no longer used - so requesting changes for now.

kantera800 commented on agent/src/beerocks/monitor/monitor_thread.cpp:
```diff
@@ -677,6 +677,10 @@ bool monitor_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
+
+        // Mark the enable flag as "false" to force update in hostapd status.
+        // The status is polled every "MONITOR_DB_AP_POLLING_RATE_SEC" and update the value.
+        mon_db.set_hostapd_enabled(false);
```
I would prefer to avoid this ugly w/a .
Even if it means removing  the operational_state checking in the son_slave and reading the configuration twice.
Remember this happens only in init and re-connection states.

morantr replied:
You are welcome to suggest a better suggestion.
I didn't want to add a message for that and this is a quite simple solution with comments explaining it.

morantr committed:
platform_manager: Remove conf_initialized flag

The "conf_initialized" is responsible for reading the common configurations only
once as long there was no change in the operational state.
Remove this flag because it solves some it causing some issues which the configuration
is not being updated if there is no change in the operational flag on the platform
manager and it allows removing a lot of irrelevant code which responsible
on updating the bpl on the ifaces status.
For now, we will read the common configuration twice but when we will move to
unified agent, we will read it only once.
See issue: https://github.com/prplfoundation/prplMesh/issues/435

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
platform_manager: Remove iface and operational status

Remove all iface and operational status members and functions.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
agent: Remove iface and operational status to platform manager

Remove all iface and operational status members and functions which participates
on sending the status to the platform manager.
As part of this commit, moved the operational status to the controller to
the hostap status handler.
Currently, it is not an optimal place, since both son_slave will send the notification,
but since this feature was disabled until now and  in the future, we will have unified
Agent, it is Ok to keep it there.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
tlvf: Remove unused status notification messages

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
tlvf: Remove status messages Auto-generated code

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
bpl: Remove iface status

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
agent: Remove iface status notification from configuration file

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
cmake: Remove BEEROCKS_NOTIFY option

This option is no longer used.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

morantr committed:
monitor: Force hostap status update on SON_CONFIG_UPDATE

When the son_slave receives HOSTAP_STATUS_CHANGED_NOTIFICATION, it
sends the controller operational notification only if the son_slave
is in an operational state.
There is a race situation which the monitor might send the status message before
the son_slave has changed to an operational state, and therefore the status update
will be sent on an irrelevant time, when the son_slave is yet not on operational
state, and/or there is no link to the controller.
To solve it, set the enable state of the hostapd on the mon_db to false when
receiving SON_CONFIG_UPDATE update message, which is sent right before the son_slave
is switching to an operational state and that way on the next status polling,
an update will be sent to the son_slave when it is opn operational state and has
a link to the controller.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

kantera800 reviewed (approved):
Forcing  hostap status update to prevent possible race between controller/son_slave is not optimal solution. 
There should be a flag in son_slave to indicate if there is a link to the controller.
But this is not related to this PR so i'm approving.
